frontend/packages/topology/src/components/side-bar/providers/TabSection.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import { GraphElement, isEdge, observer } from '@patternfly/react-topology';
+import { useTranslation } from 'react-i18next';
+import { DetailsTab, DetailsTabSection, ResolvedExtension } from '@console/dynamic-plugin-sdk';
+import { Tab } from '@console/internal/components/utils';
+import { orderExtensionBasedOnInsertBeforeAndAfter } from '@console/shared';
+import { getResource } from '@console/topology/src/utils';
+import { DefaultResourceSideBar } from '../DefaultResourceSideBar';
+import TopologyEdgeResourcesPanel from '../TopologyEdgeResourcesPanel';
+
+type TabSection = Omit<
+  DetailsTabSection['properties'] & { resolvedSection: React.ReactNode },
+  'section' | 'tab'
+>;
+
+type TabSectionProps = {
+  element: GraphElement;
+  callback: (tabs: Tab[], loaded: boolean) => React.ReactElement;
+  tabSectionExtensions: ResolvedExtension<DetailsTabSection>['properties'][];
+  tabExtensions: DetailsTab['properties'][];
+};
+
+const TabSection: React.FC<TabSectionProps> = ({
+  element,
+  callback,
+  tabSectionExtensions,
+  tabExtensions,
+}) => {
+  const { t } = useTranslation();
+
+  // resolving hooks in loop since number of extensions will remain the same
+  const tabSections: { [key: string]: TabSection[] } = tabSectionExtensions.reduce(
+    (acc, { section, tab, ...rest }) => {
+      const [resolvedSection] = section(element);
```

Comment:
Got an error here when opening a Deployment sidebar.

```
Uncaught TypeError: section is not a function or its return value is not iterable
    at dev-console-topology~overview-list-page-3b3953f124f27129bdab.js:3648:31
    at Array.reduce (<anonymous>)
    at TabSection (dev-console-topology~overview-list-page-3b3953f124f27129bdab.js:3641:44)
```

In my case section is from `observe-tab-section.tsx`, which has two cases when it returns undefined and one where it returns an element, not a resolvedSection. Can you take a look?

Reply:
None of the section methods returns an array. They either return an element or undefined. To implement them as `ExtensionHook` which returns an array with loading and error state every method will need to be changed to return an array instead of element/undefined.

Reply:
It would be also great if we can have a fallback here.

```suggestion
      let hookResult: ExtensionHookResult<React.ReactElement | undefined, GraphElement> = section(element);
      if (!Array.isArray(hookResult)) {
        console.warn("Deprected hook result for console.topology/details/tab-section " ... do we have an ID here?);
      }
      const [resolvedSection, resolved] = hookResult;
```

Wdy?